templates/bot/js/notification-function-base/README.md

```diff
@@ -1,104 +1,274 @@
+Run your app with local debugging by pressing `F5` in VSCode. Select `Debug (Edge)` or `Debug (Chrome)`.
 
-Then, you can quickly start local debugging via `F5` in VSCode. Select `Debug (Edge)` or `Debug (Chrome)` debug option of your preferred browser.
+**Congratulations**! You are running an application that can now send notifications to Teams.
 
-> **Note**: This app will setup [Microsoft Bot Framework](https://dev.botframework.com/) or [Azure Bot Service](https://azure.microsoft.com/services/bot-services/) for further running.
+## Test a http trigger
+
+If you selected `http` trigger, you can test it:
+
+* Send a POST request to `http://<endpoint>/api/notification` with your favorite tool (like `Postman`)
+  * When your project is running locally, replace `<endpoint>` with `localhost:3978`
+  * When your project is deployed to Azure Functions, replace `<endpoint>` with the url from Azure Functions
+
+>
+> **Prerequisites**
 >
-> If your account has no access to such resource(s), there's an alternative way to send notification via **Incoming Webhook**.
+> To run locally, you will need:
+>
+> - `Node.js` installed locally (recommended version: 14)
+> - An [M365 account for development](https://docs.microsoft.com/microsoftteams/platform/toolkit/accounts)
 >
-> Try the Incoming Webhook sample via `Create a new Teams app` -> `Start from a sample` -> `Incoming Webhook Notification`. Or browse the code at our [Sample Repo](https://github.com/OfficeDev/TeamsFx-Samples), `incoming-webhook-notification` folder.
 
-## Develop
+# Understanding the code
 
-This new project folder structure looks like:
+This section walks through the generated code. The project folder contains the following:
 
 | Folder | Contents |
 | - | - |
 | `.fx` | Project level settings, configurations, and environment information |
 | `.vscode` | VSCode files for local debug |
-| `bot` | The bot source code |
-| `templates` |Templates for Teams app manifest and corresponding Azure resources|
+| `bot` | The source code for the notification Teams application |
+| `templates` | Templates for the Teams application manifest and for provisioning Azure resources |
+
+The notification implementation is in `bot` folder.
+
+The following files provide the business logic for notifications. These files can be updated to fit your business logic requirements. The default implementation provides a starting point to help you get started.
+
+| File | Contents |
+| - | - |
+| `*Trigger/function.json` | Azure Function bindings for the notification trigger |
+| `src/*Trigger.js` | Notification trigger implementation |
+| `src/adaptiveCards/notification-default.json` | A generated Adaptive Card that is sent to Teams |
+| `src/cardModels.js` | The default Adaptive Card data model |
+
+The following files implement the core notification on the Bot Framework. You generally will not need to customize these files.
```

Comment:
We still want to developers to customize the storage. (it is actually a must-have).